src/components/product.js

```diff
@@ -461,7 +477,7 @@ export default class Product extends Component {
    * @return {String}
    */
   get onlineStoreURL() {
-    const identifier = this.handle ? this.handle : this.id;
+    const identifier = this.handle ? this.handle : atob(this.id).split('/').pop();
```

Comment:
also, we could actually use the same logic we use for blogs to show and hide the value in the API, as opposed to assuming it's always there generating it on the client.